docs/en/flash.md

```diff
@@ -46,28 +53,34 @@ Otherwise, if you built your own firmware from source code:
 
-If there is no init data block found during SDK startup, the SDK will install one itself. If there is a previous (potentially too old) init block, the SDK *probably* doesn't do anything with it but there is no documentation from Espressif on this topic.
+## SDK Init Data
 
-Hence, there are two strategies to update the SDK init data:
+NodeMCU versions are compiled against specific versions of the Espressif SDK.  The SDK reserves space in flash that is used to store calibration and other data.  This data changes between SDK versions, and if it is invalid or not present, the code may not boot at all.  Symptoms include messages like `rf_cal[0] !=0x05,is 0xFF`, or endless reboot loops.
 
-- Erase flash completely. This will also erase the (Lua) files you uploaded to the device! The SDK will install the init data block during startup.
-- Don't erase the flash but replace just the init data with a new file during the flashing procedure. For this you would download [SDK patch 1.5.4.1](http://bbs.espressif.com/download/file.php?id=1572) and extract `esp_init_data_default.bin` from there.
+Some earlier SDK versions would automatically populate this space if booted on an erased chip.  This is no longer the case for 1.5.4.1, where it must be flashed manually.  The data that needs to be written is provided as part of the Espressif SDK, in the file `esp_init_data_default.bin`.  For 1.5.4.1, the file can be obtained by downloading [SDK patch 1.5.4.1](http://bbs.espressif.com/download/file.php?id=1572) and extracting `esp_init_data_default.bin` from there.
 
-When flashing a new firmware (particularly with a much different size), the flash filesystem may be reformatted as the firmware starts. If it is not automatically reformatted, then it should be valid and have the same contents as before the flash operation. You can still run [`file.format()`](modules/file.md#fileformat) to re-format your flash filesystem. You will know if you need to do this if your flash files exist but seem empty, or if data cannot be written to new files. However, this should be an exceptional case.
+This file must be written to the 4th-last sector in flash.  Sectors are 4&nbsp;kB and the location will depend on the size of your flash memory.  For example, the correct address is:
 
-**esptool.py**
+- `0x7c000` for 512 kB, modules like ESP-01, -03, -07 etc.
+- `0xfc000` for 1 MB, modules like ESP8285, PSF-A85
+- `0x1fc000` for 2 MB
+- `0x3fc000` for 4 MB, modules like ESP-12E, NodeMCU devkit 1.0, WeMos D1 mini
+
+If you're not 100% sure what flash size you have, try other addresses.  Sometimes module manufacturers install a larger flash chip than they advertise.
```

Comment:
Agreed. I really want the esp_init_data_default to be completely hidden from the user by default and have NodeMCU just Do The Right Thing automatically. Which, hopefully, we have in 1526. A user laying down custom init data should be *optional*, it's far too much a hassle otherwise, as @marcelstoer has discovered.